src/rust/src/packed_r_tree.rs

```diff
@@ -570,8 +582,41 @@ impl PackedRTree {
                         offset: node_item.offset as usize,
                         index: pos - leaf_nodes_offset,
                     });
-                } else {
-                    queue.push(Reverse((node_item.offset as usize, level - 1)));
+                    continue;
+                }
+
+                // spend this many extra bytes if it means we'll eliminate an extra request
+                let combine_request_threshold = 256 * 1024 / size_of::<NodeItem>();
+
+                match (queue.back_mut(), node_item.offset as usize) {
```

Comment:
Boah, this is complicated. I trust you (and the tests), that the traversal logic is still the same. And we still have the non-optimized search methods to understand the algorithm.

Reply:
This is indeed the crux of the change and the barer of the bulk of complexity.

I'll add a high level comment to explain, as I should have done initially...

But basically:

Previously, each node traversed in the index tree was added separately to the search queue.

Now, rather than adding a new node, we'll merge it with an existing node provided:
1. there's already a node for that level of the index
2. that node is "reasonably" close (e.g if I somehow need the fourth node in a level and the three-millionth node in the level I probably don't want to merge those into a single request because I'll need to fetch all the items in-between)

Reply:
Hopefully this is a little clearer with https://github.com/bjornharrtell/flatgeobuf/pull/93/commits/eedd2df84f4093d797d6e87d3129fe14a732d87d

I combined the second and third cases and tried to clarify that the only difference is what they log.